inc/query/query.php

```diff
@@ -160,111 +162,129 @@ public function handle_query_sibling( &$clauses, $id_column ) {
 	 * @param string $clauses   Query clauses.
 	 * @param array  $args   $WP_query args object.
 	 */
-	public function filter_query_clauses( &$clauses, $args ) {
-		global $wpdb;
-		if ( 1 >= count( $args ) ) {
-			return;
-		}
-		$relationships          = $args;
-		$relationships_operator = $clauses['relation'];
-		$join_on_clauses        = array();
-		$in_post_id             = array();
+	public function handle_multiple_relationship( &$clauses, $args ) {
+		$relationships = $args;
+		$objects       = array();
+		$object_ids    = array();
 		foreach ( $relationships as $relationship ) {
-
-			$post_id = $this->get_posts_id( $relationship );
-			if ( empty( $post_id ) || 'OR' === $relationships_operator ) {
-				$in_post_id = array_merge( $in_post_id, $post_id );
-			} else {
-				$in_post_id = array_intersect( $in_post_id, $post_id );
+			$object = $this->get_relationship_object_ids( $relationship );
+			if ( $object ) {
+				$object_ids[] = $object;
 			}
-			$join_on_clause = $this->get_join_clause( $relationship );
-			if ( null !== $join_on_clause ) {
-				$join_on_clauses[] = $join_on_clause;
+
+			$object = $this->get_relationship_objects( $relationship );
+			if ( null !== $object ) {
+				$objects[] = $object;
 			}
 		}
-		if ( $in_post_id ) {
-			$this_statement_sql = implode( ',', $in_post_id );
-			$this_statement_sql = " AND $wpdb->posts.ID IN(${this_statement_sql})";
-			$clauses['where']  .= $this_statement_sql;
+		if ( $object_ids ) {
+			$this->alter_where_clause( $clauses, $object_ids );
+		}
+		if ( $objects ) {
+			$this->alter_join_clause( $clauses, $objects );
 		}
-		$tag_beginning = 'AND ((';
-		$pos           = strpos( $clauses['join'], $tag_beginning );
-		$pos           = false === $pos ? strlen( "INNER JOIN $wpdb->mb_relationships AS mbr ON" ) : $pos;
-		if ( $join_on_clauses && false !== $pos ) {
-			$clauses['join']  = substr( $clauses['join'], 0, $pos + strlen( $tag_beginning ) - 2 );
-			$clauses['join'] .= implode( " $relationships_operator ", $join_on_clauses );
-			$clauses['join'] .= ')';
+	}
+
+
+	public function alter_where_clause( &$clauses, $object_ids ) {
+		global $wpdb;
+		$mb_relatioship_type = $clauses['relation'];
+		$merge_object_ids    = array_shift( $object_ids );
+		foreach ( $object_ids as $object ) {
+			$merge_object_ids = 'OR' === $mb_relatioship_type
+				? array_merge( $merge_object_ids, $object )
+				: array_intersect( $merge_object_ids, $object );
```

Comment:
$merge_object_ids khi OR lấy phần tổng hợp, AND lấy phần chung, vì điều kiện AND là chung giữa 2 vế